scripts/release_info.py

```diff
@@ -292,6 +310,188 @@ def _check_storage_models(current_release):
+        repo: github.Repository.Repository. The PyGithub object for the repo.
+
+    Returns:
+        bool. Whether all pull requests for current release have a
+            PR: released label.
+    """
+    all_prs = repo.get_pulls(state='all')
+    for pr in all_prs:
+        label_names = [label.name for label in pr.labels]
+        if 'PR: released' not in label_names and (
+                'PR: for current release' in label_names):
+            return False
+    return True
+
+
+def update_sorted_file(filepath, new_list, start_index_line):
+    """Updates the files AUTHORS and CONTRIBUTORS with a sorted list of
+    new authors or contributors.
+
+    Args:
+        filepath: str. The path of the file to update.
+        new_list: list(str). The list of new authors or contributors to
+            add to the file.
+        start_index_line: str. The line whose index determines the start
+            of authors/contributors list in the file.
+    """
+    content = []
+    with python_utils.open_file(filepath, 'r') as f:
+        content = f.readlines()
+    start_index = content.index(start_index_line) + 2
+    updated_list = new_list + content[start_index:]
+    updated_list = sorted(updated_list, key=lambda s: s.lower())
+    content = content[:start_index] + updated_list
+    with python_utils.open_file(filepath, 'w') as f:
+        for line in content:
+            f.write(line)
+
+
+def update_changelog(release_summary_content, current_release_version):
+    """Updates CHANGELOG file.
+
+    Args:
+        release_summary_content: list(str). List of lines in
+            ../release_summary.md.
+        current_release_version: str. The version of current release.
+    """
+
+    current_date = datetime.date.today().strftime('%d %b %Y')
+    python_utils.PRINT('Updating Changelog...')
+    start_index = release_summary_content.index('### Changelog:\n') + 1
+    end_index = release_summary_content.index('### Commit History:\n')
+    release_version_changelog = [
+        u'v%s (%s)\n' % (current_release_version, current_date),
+        u'------------------------\n'] + release_summary_content[
+            start_index:end_index]
+    changelog_content = []
+    with python_utils.open_file(
+        CHANGELOG_FILEPATH, 'r') as changelog_file:
+        changelog_content = changelog_file.readlines()
+    changelog_content[2:2] = release_version_changelog
+    with python_utils.open_file(
+        CHANGELOG_FILEPATH, 'w') as changelog_file:
+        for line in changelog_content:
+            changelog_file.write(line)
+    python_utils.PRINT('Updated Changelog!')
+
+
+def update_authors(release_summary_content):
+    """Updates AUTHORS file.
+
+    Args:
+        release_summary_content: list(str). List of lines in
+            ../release_summary.md.
+    """
+    python_utils.PRINT('Updating Authors...')
+    start_index = release_summary_content.index(
+        '### New Authors:\n') + 1
+    end_index = release_summary_content.index(
```

Comment:
Same note about adding tests to ensure these sections are adjacent in the release summary file. In those backend tests, add a comment about why the check is placed there (it is so that this script can run).

Reply:
Done.